plugins/modules/ecs_cluster.py

```diff
@@ -177,12 +266,50 @@ def main():
 
     results = dict(changed=False)
     if module.params['state'] == 'present':
+        # Pull requested and existing capacity providers and strategies.
+        purge_capacity_providers = module.params['purge_capacity_providers']
+        requested_cp = module.params['capacity_providers']
+        requested_cps = module.params['capacity_provider_strategy']
+        existing_cp = existing['capacityProviders']
+        existing_cps = existing['defaultCapacityProviderStrategy']
         if existing and 'status' in existing and existing['status'] == "ACTIVE":
-            results['cluster'] = existing
+            if requested_cp is None:
```

Comment:
the `existing` variable doesn't exist when the ecs cluster does not exist (_means also nor active nor inactive_) and a `capacity_provider` is also requested.

```diff
 homepage: https://github.com/ansible-collections/community.aws
diff --git a/plugins/modules/ecs_cluster.py b/plugins/modules/ecs_cluster.py
index e54f9449..f342832d 100644
--- a/plugins/modules/ecs_cluster.py
+++ b/plugins/modules/ecs_cluster.py
@@ -270,9 +270,9 @@ def main():
         purge_capacity_providers = module.params['purge_capacity_providers']
         requested_cp = module.params['capacity_providers']
         requested_cps = module.params['capacity_provider_strategy']
-        existing_cp = existing['capacityProviders']
-        existing_cps = existing['defaultCapacityProviderStrategy']
         if existing and 'status' in existing and existing['status'] == "ACTIVE":
+            existing_cp = existing.get('capacityProviders', [])
+            existing_cps = existing.get('defaultCapacityProviderStrategy', [])
             if requested_cp is None:
                 requested_cp = []
```

see this taceback

```
The full traceback is:
Traceback (most recent call last):
  File "/home/m/.ansible/tmp/ansible-tmp-1674032324.3621714-122154-158551089939616/AnsiballZ_ecs_cluster.py", line 107, in <module>
    _ansiballz_main()
  File "/home/m/.ansible/tmp/ansible-tmp-1674032324.3621714-122154-158551089939616/AnsiballZ_ecs_cluster.py", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/m/.ansible/tmp/ansible-tmp-1674032324.3621714-122154-158551089939616/AnsiballZ_ecs_cluster.py", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.community.aws.plugins.modules.ecs_cluster', init_globals=dict(_module_fqn='ansible_collections.community.aws.plugins.modules.ecs_cluster', _modlib_path=modlib_path),
  File "/usr/lib/python3.10/runpy.py", line 224, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_ecs_cluster_payload_buos_n3u/ansible_ecs_cluster_payload.zip/ansible_collections/community/aws/plugins/modules/ecs_cluster.py", line 354, in <module>
  File "/tmp/ansible_ecs_cluster_payload_buos_n3u/ansible_ecs_cluster_payload.zip/ansible_collections/community/aws/plugins/modules/ecs_cluster.py", line 273, in main
AttributeError: 'NoneType' object has no attribute 'get'
fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/home/m/.ansible/tmp/ansible-tmp-1674032324.3621714-122154-158551089939616/AnsiballZ_ecs_cluster.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/home/m/.ansible/tmp/ansible-tmp-1674032324.3621714-122154-158551089939616/AnsiballZ_ecs_cluster.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/m/.ansible/tmp/ansible-tmp-1674032324.3621714-122154-158551089939616/AnsiballZ_ecs_cluster.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.community.aws.plugins.modules.ecs_cluster', init_globals=dict(_module_fqn='ansible_collections.community.aws.plugins.modules.ecs_cluster', _modlib_path=modlib_path),\n  File \"/usr/lib/python3.10/runpy.py\", line 224, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.10/runpy.py\", line 96, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.10/runpy.py\", line 86, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_ecs_cluster_payload_buos_n3u/ansible_ecs_cluster_payload.zip/ansible_collections/community/aws/plugins/modules/ecs_cluster.py\", line 354, in <module>\n  File \"/tmp/ansible_ecs_cluster_payload_buos_n3u/ansible_ecs_cluster_payload.zip/ansible_collections/community/aws/plugins/modules/ecs_cluster.py\", line 273, in main\nAttributeError: 'NoneType' object has no attribute 'get'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```